src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -169,6 +194,34 @@ public static BicepCompletionContext Create(IFeatureProvider featureProvider, Co
                 activeScopes);
         }
 
+        private static bool IsDisableNextLineDiagnosticsDirectiveStartContext(BicepFile bicepFile, int offset, List<SyntaxBase> matchingNodes)
+        {
+            var tokens = matchingNodes.Where(x => x is Token token &&
+                                             token.Text == "#" &&
+                                             token.Span.GetEndPosition() == offset &&
+                                             !TextBetweenLineStartAndTokenContainsNonWhiteSpaceCharacter(bicepFile, token));
+
+            return tokens.Count() == 1;
+        }
+
+        private static bool TextBetweenLineStartAndTokenContainsNonWhiteSpaceCharacter(BicepFile bicepFile, Token token)
+        {
+            var lineStarts = bicepFile.LineStarts;
+            var position = token.GetPosition();
+            (var line, _) = TextCoordinateConverter.GetPosition(lineStarts, position);
+            var lineStart = lineStarts[line];
+
+            for (int i = lineStart; i < position; i++)
```

Comment:
I don't think this approach works for comments - I get completions with:

```bicep
/* asdfasdf */#|
```

You probably will also need to check the token for leading trivia syntax of type comment to handle this case.

Reply:
BTW - we've talked about tightening this feature up so that `#` has to be at the start of the line. Would that make this check simpler?

Reply:
Sure, updated this to add check to make sure token is at lineStart:
https://github.com/Azure/bicep/commit/73876114b00d691fdb61cca0ffd938ad7ec585a8